src/EditorFeatures/CSharp/NavigationBar/CSharpEditorNavigationBarItemService.cs

```diff
@@ -27,26 +25,9 @@ public CSharpEditorNavigationBarItemService(IThreadingContext threadingContext)
         {
         }
 
-        protected override async Task<VirtualTreePoint?> GetSymbolNavigationPointAsync(
-            Document document, ISymbol symbol, CancellationToken cancellationToken)
+        protected override async Task<bool> TryNavigateToItemAsync(Document document, WrappedNavigationBarItem item, ITextView textView, ITextSnapshot textSnapshot, CancellationToken cancellationToken)
         {
-            var syntaxTree = await document.GetRequiredSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
-            var location =
-                symbol.Locations.FirstOrDefault(l => Equals(l.SourceTree, syntaxTree)) ??
-                symbol.Locations.FirstOrDefault(l => l.SourceTree != null);
-
-            if (location == null)
-                return null;
-
-            var tree = location.SourceTree;
-            Contract.ThrowIfNull(tree);
-
-            return new VirtualTreePoint(tree, tree.GetText(cancellationToken), location.SourceSpan.Start);
-        }
-
-        protected override async Task<bool> TryNavigateToItemAsync(Document document, WrappedNavigationBarItem item, ITextView textView, CancellationToken cancellationToken)
-        {
-            await NavigateToSymbolItemAsync(document, (RoslynNavigationBarItem.SymbolItem)item.UnderlyingItem, cancellationToken).ConfigureAwait(false);
+            await NavigateToSymbolItemAsync(document, item, (RoslynNavigationBarItem.SymbolItem)item.UnderlyingItem, textSnapshot, cancellationToken).ConfigureAwait(false);
             return true;
```

Comment:
c# no longer needs to do any sort of complex processing logic when navigating.  VB still does, but it's much more minor (And can hopefully be removed ina  followup).